src/baltree.h

```diff
@@ -45,26 +60,76 @@
 #ifndef BALANCED_TREE_INIT    // non-generic part shared by all versions
 #define BALANCED_TREE_INIT
 
+// Scapegoat trees are height-balanced trees. With respect to a
+// parameter alpha, a binary tree is height-balanced, if
+//
+//   height(tree) <= log_1/alpha(size(tree))
+//
+// where size(tree) is the number of nodes that a tree contains.
+//
+// A binary tree is called weight-balanced iff for any subtree `tree`,
+//
+//   size(left) <= alpha * size(tree) and
+//   size(right) <= alpha * size(tree)
+//
+// where `left` and `right` are the left and right subtrees of `tree`.
```

Comment:
So `size(left)+size(right)+1 = size(tree)` -- I guess that means alpha actually must be between 0.5 and 1 or so?